MainWindow.xaml.cs

```diff
@@ -13,16 +13,232 @@
+                    {
+                        int idx = 0;
+                        foreach (TextBox item in textBoxes)
+                        {
+                            ++idx;
+                            if (item == textBox)
+                            {
+                                textBoxes[idx].Focus();
+                                if (textBoxes[idx].Text.Length > 0)
+                                {
+                                    textBoxes[idx].SelectAll();
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        textBox.Text = "";
+                    }
+                }
+                else
+                {
+                    textBox.Text = "";
+                }
+            }
+            else if (textBox.Text.Length > 2)
+            {
+                textBox.Text = textBox.Text.Substring(0, 2);
+            }
+
+            if (txtHStart.Text.Length == 2 && txtHStartPause.Text.Length == 2 && txtHEndPause.Text.Length == 2 &&
+                txtMStart.Text.Length == 2 && txtMStartPause.Text.Length == 2 && txtMEndPause.Text.Length == 2)
+            {
+                hStart = int.Parse(txtHStart.Text);
+                mStart = int.Parse(txtMStart.Text);
+                hStartPause = int.Parse(txtHStartPause.Text);
+                mStartPause = int.Parse(txtMStartPause.Text);
+                hEndPause = int.Parse(txtHEndPause.Text);
+                mEndPause = int.Parse(txtMEndPause.Text);
+
+                calculateEndPause();
+            }
+        }
+
+        private void txt_TextSuppChanged(object sender, TextChangedEventArgs e)
+        {
+            if (txtHEnd.IsReadOnly)
+            {
+                if ((txtHTimeSupp.Text.Length == 2 && !txtHTimeSupp.Text.StartsWith("-")) || txtHTimeSupp.Text.Length == 3 && txtHTimeSupp.Text.StartsWith("-"))
+                {
+                    txtMTimeSupp.Focus();
+                }
+                if (txtHEnd.Text.Length == 2 &&
+                    ((txtHTimeSupp.Text.Length == 2 && !txtHTimeSupp.Text.StartsWith("-")) ||
+                    (txtHTimeSupp.Text.Length == 3 && txtHTimeSupp.Text.StartsWith("-"))) &&
+                    txtMTimeSupp.Text.Length == 2)
+                {
+                    calculateHSupp();
+                }
+            }
+        }
+
+        private void calculateEndPause()
+        {
+            int hRemaining, mRemaining;
+            int hMatin, mMatin;
+
+            hMatin = hStartPause - hStart;
+            if (mStartPause < mStart)
+            {
+                --hMatin;
+                mMatin = 60 - mStart + mStartPause;
+            }
+            else
+            {
+                mMatin = mStartPause - mStart;
+            }
+
+            hRemaining = 8 - hMatin;
```

Comment:
Shouldn't we output the daily time in a key in the AppConfig file?
We all have different times to do things. Example: I have to do 8.5 hours a day, someone else has to do 8.24 or 8h a day, etc.

Reply:
or at least in a constant or by user choice

Reply:
It will be done in the code clean-up in another branch, this is only an importation